### ISSUE TITLE
🐛 Bug: Fail to identify README.zh-CN.md file for no-redundant-files rule

### ISSUE THREAD

**Issue by chouchouji** [JoshuaKGoldberg/eslint-plugin-package-json#763]
### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have pulled the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

I run `npm pack` in terminal. It includes all readme files.

![Image](https://github.com/user-attachments/assets/dd39672a-49ca-44b9-b015-95db70b48f23)

But when I set files to 

```js
  "files": [
    "lib",
    "README.md",
    "README.zh-CN.md"
  ],
```

only the `README.md` is reported.

![Image](https://github.com/user-attachments/assets/087306e8-cfa0-4743-adf5-c26e2184178b)

### Actual

The `README.zh-CN.md` isn't recognized by this plugin.

### Additional Info

Should we use `glob` or other tools to identify files?

![Image](https://github.com/user-attachments/assets/1b461a9a-1dc5-4c4d-9a5e-786d8178c71b)

**Comment by michaelfaith** [comment]
Interesting. https://docs.npmjs.com/cli/v9/configuring-npm/package-json#files Doesn't really give any details about how forgiving it is on this match.  As in, will it accept _anything_ after `README.`?  I'd hate to flag something with the rule as being redundant when it actually isn't.

**Comment by JoshuaKGoldberg** [comment]
Good find, thanks for reporting! It does seem like an ambiguity in the docs that they don't clarify what _"extension"_ means.

I tried it out in https://github.com/JoshuaKGoldberg/repros/tree/npm-publish-readme-multi-period-extensions and confirmed that, yes, files like `README.zh-CN.md` are included by default. npm does accept anything after `README.`.

Btw @chouchouji please don't ignore the accessibility bot. Posting images without alt text really is not good: folks who can't visually see the images won't be able to read their contents. They're also less searchable in GitHub issues and on search engines. The highlighting you added to them is helpful (and appreciated!), but it'd be better to instead copy and paste links and text.

For example, referencing the npm docs, https://docs.npmjs.com/cli/configuring-npm/package-json#files:

> `README` & `LICENSE` can have any case and extension.

**Comment by chouchouji** [comment]
> Good find, thanks for reporting! It does seem like an ambiguity in the docs that they don't clarify what _"extension"_ means.
> 
> I tried it out in https://github.com/JoshuaKGoldberg/repros/tree/npm-publish-readme-multi-period-extensions and confirmed that, yes, files like `README.zh-CN.md` are included by default. npm does accept anything after `README.`.
> 
> Btw [@chouchouji](https://github.com/chouchouji) please don't ignore the accessibility bot. Posting images without alt text really is not great: folks who can't visually see the images won't be able to read their contents. They're also less searchable in GitHub issues and on search engines. The highlighting you added to them is helpful (and appreciated!), but it'd be better to instead copy and paste links and text.
> 
> For example, referencing the npm docs, https://docs.npmjs.com/cli/configuring-npm/package-json#files:
> 
> > `README` & `LICENSE` can have any case and extension.

I get it. Thanks for you confirm. Maybe we should check `npm cli` code to get the real pattern.

**Comment by chouchouji** [comment]
> Interesting. Nice find. https://docs.npmjs.com/cli/v9/configuring-npm/package-json#files Doesn't really give any details about how forgiving it is on this match. As in, will it accept _anything_ after `README.`? I'd hate to flag something with the rule as being redundant when it actually isn't.

I think they are `README*` and `LICENSE*`.

**Comment by chouchouji** [comment]
> Good find, thanks for reporting! It does seem like an ambiguity in the docs that they don't clarify what _"extension"_ means.
> 
> I tried it out in https://github.com/JoshuaKGoldberg/repros/tree/npm-publish-readme-multi-period-extensions and confirmed that, yes, files like `README.zh-CN.md` are included by default. npm does accept anything after `README.`.
> 
> Btw [@chouchouji](https://github.com/chouchouji) please don't ignore the accessibility bot. Posting images without alt text really is not great: folks who can't visually see the images won't be able to read their contents. They're also less searchable in GitHub issues and on search engines. The highlighting you added to them is helpful (and appreciated!), but it'd be better to instead copy and paste links and text.
> 
> For example, referencing the npm docs, https://docs.npmjs.com/cli/configuring-npm/package-json#files:
> 
> > `README` & `LICENSE` can have any case and extension.

I get it. Thanks for your confirm. Maybe we should check `npm cli` code to find real pattern.
